src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -278,51 +280,58 @@ public void write(List<Object> values) {
                 + ") : "
                 + values);
       }
-
       recordConsumer.startMessage();
+      writeValues(values, schema);
+      recordConsumer.endMessage();
+    }
+
+    private void writeValues(List<Object> values, GroupType type) {
+      List<Type> cols = type.getFields();
       for (int i = 0; i < cols.size(); ++i) {
         Object val = values.get(i);
-        // val.length() == 0 indicates a NULL value.
         if (val != null) {
-          String fieldName = cols.get(i).getPath()[0];
+          String fieldName = cols.get(i).getName();
```

Comment:
whats the difference between getPath()[0] and getName() ?

Reply:
The `getPath()[0]` always returns the root column name. While `getName()` returns the current column/subcolumn name. This does matters for primitive data type column. Is needed for structured data type.